app/models/piece.rb

```diff
@@ -8,4 +8,91 @@ class Piece < ActiveRecord::Base
   scope :rooks, -> { where(type: 'Rook') }
   scope :queens, -> { where(type: 'Queen') }
   scope :kings, -> { where(type: 'King') }
+
+  def is_obstructed?(row, column)
+    if moving_horizontally?(row)
+      is_horizontal_obstructed?(column)
+    elsif moving_vertically?(column)
+      is_vertical_obstructed?(row)
+    elsif moving_diagonally?(row, column)
+      is_diagonal_obstructed?(row, column)
+    else
+      fail 'Input is invalid!'
+    end
+  end
```

Comment:
another way to write this is with guard statements. it would look like this.

```Ruby
  def is_obstructed?(row, column)
    return horizontal_obstructed?(row) if moving_horizontally?(row)
    return vertical_obstructed?(row) if moving_vertically?(column)
    return diagonal_obstructed?(row, column) if moving_diagonally?(row, column)

    fail 'Input is invalid!'
 end
```